File: test-output/python/test_multi_cli/cli_hooks.py
"""
Hook implementations for Multi-Language Test CLI.

This file contains the business logic for your CLI commands.
Implement the hook functions below to handle your CLI commands.

IMPORTANT: Hook names must use snake_case with 'on_' prefix
Example:
- Command 'hello' -> Hook function 'on_hello'
- Command 'hello-world' -> Hook function 'on_hello_world'
"""

# Import any modules you need here
import sys
import json
from typing import Any, Dict, Optional
def on_greet(    enthusiastic: Optional[bool] = None,    **kwargs
) -> Dict[str, Any]:
    """
    Handle greet command.        enthusiastic: Be enthusiastic    
    Returns:
        Dictionary with status and optional results
    """
    # Add your business logic here
    print(f"Executing greet command")    
    return {
        "status": "success",
        "message": "greet completed successfully"
    }
def on_info(    verbose: Optional[bool] = None,    **kwargs
) -> Dict[str, Any]:
    """
    Handle info command.        verbose: Verbose output    
    Returns:
        Dictionary with status and optional results
    """
    # Add your business logic here
    print(f"Executing info command")    
    return {
        "status": "success",
        "message": "info completed successfully"
    }

File: test-output/rust/src/cli_hooks.rs
// Hook implementations for Multi-Language Test CLI
//
// This file contains the business logic for your CLI commands.
// Implement the hook functions below to handle your CLI commands.
//
// IMPORTANT: Hook names must use snake_case with 'on_' prefix
// Example:
// - Command 'hello' -> Hook function 'on_hello'
// - Command 'hello-world' -> Hook function 'on_hello_world'

use clap::ArgMatches;
use anyhow::Result;

/// Greet someone
pub fn on_greet(matches: &ArgMatches) -> Result<()> {
    // Add your business logic here
    println!("Executing greet command");
    // Access options
    if let Some(enthusiastic) = matches.get_one::<String>("enthusiastic") {
        println!("enthusiastic: {}", enthusiastic);
    }
    
    Ok(())
}/// Show information
pub fn on_info(matches: &ArgMatches) -> Result<()> {
    // Add your business logic here
    println!("Executing info command");
    // Access options
    if let Some(verbose) = matches.get_one::<String>("verbose") {
        println!("verbose: {}", verbose);
    }
    
    Ok(())
}